drivers/soundwire/cadence_master.c

```diff
@@ -925,9 +925,23 @@ irqreturn_t sdw_cdns_irq(int irq, void *dev_id)
 	}
 
 	if (int_status & CDNS_MCP_INT_SLAVE_MASK) {
+
+		u32 slv_mask;
+
+		slv_mask = cdns_readl(cdns, CDNS_MCP_INTMASK);
+
+		/* FIXME: there must be a clever way to do this in one shot */
```

Comment:
slv_mask &= ~(int_status & CNDS_MCP_INT_SLAVE_MASK); ?

Reply:
I am not sure if it's equivalent but I wrote this down:

slv_mask &= (~int_status & CNDS_MCP_INT_SLAVE_MASK);

Reply:
not equal.  Your method will only reserve the bit in range of CNDS_MCP_INT_SLAVE_MASK, for other no-slave int status will lost